Ucommerce.Sitefinity.UI/Mvc/Model/FacetsFilterModel.cs

```diff
@@ -14,88 +14,92 @@
+		public virtual IList<FacetViewModel> CreateViewModel()
+		{
+			Ucommerce.EntitiesV2.Category currentCategory = null;
+
+			if (CatalogContext.CurrentCategory != null)
+			{
+				currentCategory = Ucommerce.EntitiesV2.Category.FirstOrDefault(c => c.Name == CatalogContext.CurrentCategory.Name);
+				return this.GetAllFacets(currentCategory);
+			}
+
+			var pageContext = SystemManager.CurrentHttpContext.GetProductsContext();
+			var categoryIds = pageContext.GetValue<ProductsController>(p => p.CategoryIds);
+			var productIds = pageContext.GetValue<ProductsController>(p => p.ProductIds);
+			if (!string.IsNullOrEmpty(categoryIds) || !string.IsNullOrEmpty(productIds))
+			{
+				return this.MapFacetsByManualSelection(categoryIds, productIds);
+			}
+
+			return this.GetAllFacets(currentCategory);
+		}
+
+		public virtual bool CanProcessRequest(Dictionary<string, object> parameters, out string message)
+		{
+			if (Telerik.Sitefinity.Services.SystemManager.IsDesignMode)
+			{
+				message = "The widget is in Page Edit mode.";
+				return false;
+			}
+
+			message = null;
+			return true;
+		}
+
+		private IList<FacetViewModel> GetAllFacets(Ucommerce.EntitiesV2.Category category)
+		{
+			var facets = HttpContext.Current.Request.QueryString.ToFacets();
+			IList<Ucommerce.Search.Facets.Facet> allFacets;
+
+			if (category != null)
+			{
+				allFacets = CatalogLibrary.GetFacets(category.Guid, facets.ToFacetDictionary());
+			}
+			else
+			{
+				allFacets = CatalogLibrary.GetFacets(CatalogContext.CurrentCatalog.Categories, facets.ToFacetDictionary());
+			}
+
+			return this.MapToFacetsViewModel(allFacets);
+		}
+
+		private IList<FacetViewModel> MapFacetsByManualSelection(string categoryIdsString, string productIdsString)
+		{
+			var categoryIds = categoryIdsString?.Split(',').Select(x => Convert.ToInt32(x)).ToList() ?? new List<int>();
+			var productIds = productIdsString?.Split(',').Select(x => Convert.ToInt32(x)).ToList() ?? new List<int>();
+			var facets = HttpContext.Current.Request.QueryString.ToFacets();
+			var categories = Ucommerce.EntitiesV2.Category.Find(x => categoryIds.Contains(x.CategoryId));
+
+			return this.MapToFacetsViewModel(
+				CatalogLibrary.GetFacets(categories.Select(x => x.Guid).ToList(), facets.ToFacetDictionary()));
+		}
+
+		private IList<FacetViewModel> MapToFacetsViewModel(IList<Ucommerce.Search.Facets.Facet> facets)
+		{
+			var facetViewModel = facets.Select(x => new FacetViewModel()
+			{
+				DisplayName = x.DisplayName,
+				Name = x.Name,
+				FacetValues = x.FacetValues
+					.Select(y => new FacetValue(y.Value, (int)y.Count))
+					.ToList(),
+			});
+
+			var currentPriceGroup = Ucommerce.EntitiesV2.PriceGroup.FirstOrDefault(x => x.Guid == CatalogContext.CurrentPriceGroup.Guid);
+
+			return facetViewModel.Where(f => f.FacetValues.Any() && (f.Name.ToLower().Contains("price") && (f.Name.ToLower().Contains("price") && f.Name.EndsWith(currentPriceGroup.Name)))).ToList();

```

Comment:
I'd love to see a unit test for this.
The commit is called `Ignored all PriceGroups that aren't the current currency` but I don't see any reference to the current currency in the code. Anyone reading the code in the future will also wonder what that line does.